File: yongkeun/8week/덧칠하기.js
function solution(n, m, section) {
  last = 0; //현재까지 칠해진 구간
  count = 0;
  section.forEach((value) => {
    //section 배열을 방문하면서
    if (value > last) {
      //현재 값이 last보다 크면(롤러질이 우측 끝을 벗어나지 않는다면)
      last = value + m - 1; //last는 현재값+롤러값-1이 됨, -1의 이유는 last가 3이면 3까지는 칠해져있다는 의미이므로 4부터 시작되야하기 때문
      count++;
    }
  });
  return count;
}

File: dongmin/week8/덧칠하기.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public int solution(int n, int m, int[] section) {
        int answer =0;
        int maxNum = 0;     //칠해진 벽면에 수가 된다.
        for (int i = 0; i < section.length; i++) {
            if (section[i] < maxNum) {      //maxNum과 section에 i번째 인덱스를 비교해서 칠해져있으면 넘어간다.
                continue;
            }
            answer++;       //칠해져있지않으면 answer를 증가시키고
            maxNum = section[i] + m;        //칠해져있지 않으면 그 벽면부터 m번까지 칠한다.
        }
        return answer;
    }
}
